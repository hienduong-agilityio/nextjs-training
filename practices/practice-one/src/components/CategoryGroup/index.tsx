'use client';

// Libraries
import { usePathname, useSearchParams } from 'next/navigation';
import { twMerge } from 'tailwind-merge';

// Components
import Link from 'next/link';

// Helpers
import { capitalizeText } from '@/helpers';

// Constants
import { ALL_CATEGORIES, SEARCH_PARAMS } from '@/constants';

interface ICategoryGroupProps {
  title: string;
  items: { name: string; count: number }[];
}

export function CategoryGroup({ title, items }: ICategoryGroupProps) {
  const pathname = usePathname();
  const searchParams = useSearchParams();

  const currentCategory =
    searchParams.get(SEARCH_PARAMS.CATEGORY) ?? ALL_CATEGORIES.ALL;

  const handleCategoryClick = (itemName: string) => {
    const updatedParams = new URLSearchParams();

    // Update the category query parameter
    if (itemName !== ALL_CATEGORIES.ALL) {
      updatedParams.set(SEARCH_PARAMS.CATEGORY, itemName);
    }

    return `${pathname}?${updatedParams.toString()}`;
  };

  return (
    <div className="mb-6 bg-secondary-100 p-4 rounded">
      <h3 className="text-lg font-bold mb-4">{title}</h3>
      <ul className="space-y-2 text-sm text-gray-700">
        {items.map((item) => {
          const isActive =
            item.name === ALL_CATEGORIES.ALL
              ? currentCategory === ALL_CATEGORIES.ALL || !currentCategory
              : item.name === currentCategory;

          return (
            <li className="flex justify-between" key={item.name}>
              <Link
                href={handleCategoryClick(item.name)}
                className={twMerge(
                  'w-full block font-semibold py-2 rounded hover:bg-secondary-200',
                  isActive
                    ? 'bg-primary-500 text-primary-200'
                    : 'text-gray-800',
                )}
              >
                {capitalizeText(item.name)}
              </Link>
            </li>
          );
        })}
      </ul>
    </div>
  );
}
