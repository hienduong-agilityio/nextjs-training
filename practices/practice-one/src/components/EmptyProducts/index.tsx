import Link from 'next/link';

export const EmptyProducts = ({
  message,
  actionLabel,
  actionHref,
}: {
  message: string;
  actionLabel: string;
  actionHref: string;
}) => (
  <div className="flex flex-col items-center gap-4 py-16">
    <p className="text-xl text-gray-600">{message}</p>
    <Link
      href={actionHref}
      className="inline-block px-6 py-2 text-white bg-blue-500 rounded-lg"
    >
      {actionLabel}
    </Link>
  </div>
);
