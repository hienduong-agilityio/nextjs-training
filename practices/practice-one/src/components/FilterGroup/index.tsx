interface FilterGroupProps {
  title: string;
  items: { name: string; count: number }[];
}

export const FilterGroup = ({ title, items }: FilterGroupProps) => {
  return (
    <div className="mb-6 bg-secondary-100 p-4 rounded">
      <h3 className="text-lg font-bold mb-4">{title}</h3>
      <ul className="space-y-2 text-sm text-gray-700">
        {items.map((item) => (
          <li key={item.name} className="flex justify-between py-2">
            <button className="font-semibold hover:text-primary-200 ">
              {item.name}
            </button>
            <span>({item.count})</span>
          </li>
        ))}
      </ul>
    </div>
  );
};
