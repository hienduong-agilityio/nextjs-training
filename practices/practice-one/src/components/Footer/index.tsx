// Icon
import {
  FacebookIcon,
  LogoIcon,
  MasterCardIcon,
  PaypalIcon,
  TwitterIcon,
  VisaIcon,
  WesternIcon,
} from '@/icons';

// Mock
import { FOOTER_NAVIGATION } from '@/mocks';

// Components
import Link from 'next/link';

// Constants
import { ROUTE } from '@/constants';

export const Footer = () => {
  return (
    <footer className="w-full bg-primary-50">
      <div className="md:container px-4 pt-6 md:mx-auto sm:px-6 md:px-8 lg:px-12 xl:px-16 sm:pt-8 md:pt-12 lg:pt-16 xl:pt-20">
        <div className="py-8">
          {/* Main Footer Content */}
          <div className="flex flex-col gap-10 lg:flex-row lg:justify-between">
            {/* Logo Section */}
            <div className="max-w-[290px] flex flex-col gap-4">
              <Link href={ROUTE.PRODUCTS} className="flex items-center gap-2">
                <LogoIcon aria-label="logo" />
                <span className="text-lg font-bold sm:text-xl lg:text-2xl">
                  E-Comm
                </span>
              </Link>
              <span className="text-sm font-normal leading-normal">
                Lorem Ipsum is simply dummy text of the printing and typesetting
                industry. Lorem Ipsum has been the industry standard dummy text
                ever. Since the 1500s, when an unknown printer.
              </span>
            </div>

            <div className="flex flex-col w-full gap-8 sm:flex-row sm:justify-evenly lg:w-3/4">
              {/* Follow Us */}
              <div className="lg:max-w-[220px] flex flex-col gap-4">
                <h4 className="font-medium text-foreground-400">Follow Us</h4>
                <span className="text-sm">
                  Since the 1500s, when an unknown printer took a galley of type
                  and scrambled.
                </span>
                <div className="flex gap-4">
                  <Link href="https://www.facebook.com/" aria-label="Facebook">
                    <FacebookIcon color="#385C8E" />
                  </Link>
                  <Link href="https://www.twitter.com/" aria-label="Twitter">
                    <TwitterIcon />
                  </Link>
                </div>
              </div>

              {/* Contact Us */}
              <div className="lg:max-w-[150px] flex flex-col gap-4">
                <h4 className="font-medium text-foreground-400">Contact Us</h4>
                <address className="text-sm not-italic">
                  E-Comm, 4578 Marmora Road, Glasgow D04 89GR
                </address>
              </div>
            </div>
          </div>

          {/* Footer Links */}
          <div className="grid grid-cols-1 gap-8 pt-16 sm:grid-cols-2 lg:grid-cols-4">
            {FOOTER_NAVIGATION.map((item) => (
              <ul key={item.title}>
                <li>
                  <h4 className="mb-5 font-medium text-foreground-400">
                    {item.title}
                  </h4>
                  <ul>
                    {item.items.map((item) => (
                      <li
                        key={item.name}
                        className="mb-3 text-sm text-content2 hover:text-primary-700"
                      >
                        <Link href={item.url} className="hover:underline">
                          {item.name}
                        </Link>
                      </li>
                    ))}
                  </ul>
                </li>
              </ul>
            ))}
          </div>
        </div>

        {/* Footer Bottom */}
        <div className="flex flex-col items-center justify-between gap-6 py-6 border-t-2 border-secondary-100 lg:flex-row ">
          <p className="text-sm text-secondary-500">
            &#169; 2018 Ecommerce theme by www.bisenbaev.com
          </p>
          <div className="flex gap-2">
            <WesternIcon aria-label="Western" />
            <MasterCardIcon aria-label="MasterCard" />
            <PaypalIcon aria-label="Paypal" />
            <VisaIcon aria-label="Visa" />
          </div>
        </div>
      </div>
    </footer>
  );
};
