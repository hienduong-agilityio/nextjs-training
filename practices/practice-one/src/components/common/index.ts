export { default as Button } from './Button';
export { default as InputField } from './InputField';
export { default as Tabs } from './Tabs';
export { default as Select } from './Select';
export { default as IconButton } from './IconButton';
export * from './IconButton';
