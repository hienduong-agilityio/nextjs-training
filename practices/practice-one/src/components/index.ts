export * from './Header';
export { default as SearchBox } from './SearchBox';
export * from './SearchBox';
export * from './ProductTab';

// Common
export * from './common';
