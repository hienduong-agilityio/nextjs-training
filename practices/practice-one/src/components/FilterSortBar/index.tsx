'use client';

// Hooks
import { useSearchParams, useRouter } from 'next/navigation';
import { useEffect, useState } from 'react';

// Components
import { Select } from '@/components';

// Constants
import { SEARCH_PARAMS, SORT_PRODUCT_OPTIONS } from '@/constants';

interface FilterSortBarProps {
  itemCount: number;
  sortOptions: string[];
  showOptions: string[];
}

export const FilterSortBar = ({
  itemCount = 0,
  sortOptions = SORT_PRODUCT_OPTIONS,
  showOptions = ['6'],
}: FilterSortBarProps) => {
  const searchParams = useSearchParams();
  const router = useRouter();

  const [selectedLimit, setSelectedLimit] = useState<string>(showOptions[0]);
  const [selectedSort, setSelectedSort] = useState<string>(sortOptions[0]);

  // TODO: Remove useEffect.
  useEffect(() => {
    // Get query parameters from the URL
    const limitParam = searchParams.get(SEARCH_PARAMS.LIMIT) ?? showOptions[0];
    const sortParam = searchParams.get(SEARCH_PARAMS.SORT_BY) ?? sortOptions[0];

    setSelectedLimit(limitParam);
    setSelectedSort(sortParam);
  }, [searchParams, showOptions, sortOptions]);

  const updateQueryParams = (param: string, value: string) => {
    const currentParams = new URLSearchParams(searchParams.toString());
    currentParams.set(param, value);

    // Reset pagination if a filter changes
    if (param === SEARCH_PARAMS.LIMIT) {
      currentParams.delete(SEARCH_PARAMS.PAGE);
    }

    router.push(`?${currentParams.toString()}`);
  };

  const handleShowChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value;

    setSelectedLimit(value);
    updateQueryParams(SEARCH_PARAMS.LIMIT, value);
  };

  const handleSortChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value;

    setSelectedSort(value);
    updateQueryParams(SEARCH_PARAMS.SORT_BY, value);
  };

  return (
    <div className="bg-secondary-100 p-4 shadow-md md:sticky rounded top-0 z-30 w-full">
      <div className="flex flex-col sm:flex-row sm:justify-between sm:items-center space-y-4 sm:space-y-0">
        {/* Item count */}
        <div className="text-sm w-max text-gray-600">{itemCount} Items</div>

        {/* Filters */}
        <div className="flex flex-col sm:flex-row sm:items-center sm:space-x-4 space-y-4 sm:space-y-0">
          <div className="flex gap-2 items-center">
            <span>Sort</span>
            {/* TODO: Custom style for select arrow by tailwinds classes*/}
            <Select
              options={sortOptions}
              value={selectedSort}
              customClass="w-auto bg-secondary-100 capitalize"
              onChange={handleSortChange}
            />
          </div>
          <div className="flex gap-2 items-center">
            <span>Show</span>
            <Select
              options={showOptions}
              value={selectedLimit}
              customClass="w-auto bg-secondary-100"
              onChange={handleShowChange}
            />
          </div>
        </div>
      </div>
    </div>
  );
};
