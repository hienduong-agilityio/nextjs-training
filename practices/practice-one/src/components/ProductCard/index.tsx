'use client';

// Libraries
import { memo } from 'react';

// Interfaces
import { IProductProps } from '@/interfaces';

// Icons
import { HeartIcon, StarRating, AddToCartIcon } from '@/icons';

// Components
import Image from 'next/image';
import { IconButton } from '@/components';
import Link from 'next/link';

// Constants
import { ROUTE } from '@/constants';

interface IProductCardProps extends Omit<IProductProps, 'images'> {
  images: string[];
  addToFavorites?: () => void;
  addToCart?: () => void;
}

const ProductCard = ({
  id,
  name = 'Product',
  images = [],
  price = 100,
  originalPrice = '50',
  discount = '50%',
  label = 'Hot',
  rating = 4,
  addToFavorites = () => {},
  addToCart = () => {},
}: IProductCardProps) => {
  const image = images.length > 0 ? images[0] : '/images/image-placeholder.svg';

  return (
    <div
      key={id}
      className="relative flex flex-col transition-shadow border-2 rounded-sm border-secondary-200 hover:shadow-lg"
    >
      {label && (
        <span className="absolute top-0 left-0 z-20 px-3 py-1 text-xs text-white uppercase rounded-sm bg-danger-200">
          {label}
        </span>
      )}
      <div className="relative bg-secondary-200 group">
        <Image
          src={image}
          alt={name}
          width={0}
          height={0}
          className="mix-blend-multiply object-contain h-[273px] w-[200px] sm:w-[300px] sm:h-[273px] mx-auto"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
        />
        {/* Hover Buttons */}
        <div className="absolute inset-0 z-10 flex items-center justify-center gap-4 mx-3 my-5 transition-opacity duration-300 bg-white opacity-0 pointer-events-none group-hover:opacity-100 group-hover:pointer-events-auto focus-within:opacity-100 focus-within:pointer-events-auto">
          <IconButton
            aria-label="Favorite product"
            customClass="flex p-2 text-sm font-medium bg-indigo-600 border-2 rounded-full text-primary-100 border-primary-100"
            onClick={addToFavorites}
          >
            <HeartIcon size={24} />
          </IconButton>
          <IconButton
            aria-label="Add to Cart"
            customClass="p-2 text-sm font-medium bg-indigo-600 border-2 rounded-full text-primary-100 border-primary-100"
            onClick={addToCart}
          >
            <AddToCartIcon size={24} />
          </IconButton>
        </div>
      </div>
      <div className="flex flex-col items-center justify-between h-full px-4 py-4 bg-white">
        <Link
          href={`${ROUTE.PRODUCT}/${id}`}
          className="text-lg font-semibold text-center text-indigo hover:underline"
        >
          {name}
        </Link>
        <StarRating size={14} rating={rating} />
        <div className="flex items-center space-x-2">
          <span className="text-lg font-medium text-primary-200">${price}</span>
          {originalPrice && (
            <span className="text-sm text-gray-400 line-through">
              ${originalPrice}
            </span>
          )}
          {discount && (
            <span className="text-sm font-semibold text-danger-50">
              ${discount}
            </span>
          )}
        </div>
      </div>
    </div>
  );
};

export default memo(ProductCard);
