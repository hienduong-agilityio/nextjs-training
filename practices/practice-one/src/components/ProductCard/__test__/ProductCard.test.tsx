// Libraries
import React from 'react';
import {
  render,
  screen,
  fireEvent,
  RenderResult,
} from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import { ProductCard } from '@/components';
import { IProductProps } from '@/interfaces';

const mockOnAddToCart = jest.fn();
const mockOnFavorite = jest.fn();

let defaultProps: IProductProps;
let renderResult: RenderResult;

describe('ProductCard Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();

    defaultProps = {
      id: '1',
      name: 'Test Product',
      images: ['/test-image.jpg'],
      price: '50',
      originalPrice: '70',
      discount: '28% OFF',
      label: 'Hot',
      rating: 4,
    };

    renderResult = render(
      <ProductCard
        {...defaultProps}
        addToFavorites={mockOnFavorite}
        addToCart={mockOnAddToCart}
      />,
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = renderResult;
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the ProductCard component with default props', () => {
    expect(screen.getByText('Test Product')).toBeInTheDocument();
    expect(screen.getByText('$50')).toBeInTheDocument();
    expect(screen.getByText('$70')).toBeInTheDocument();
    expect(screen.getByText('Hot')).toBeInTheDocument();

    const image = screen.getByAltText('Test Product');
    expect(image).toBeInTheDocument();

    const rating = screen.getAllByRole('img', { hidden: true });
    expect(rating).toHaveLength(1);
  });

  it('renders hover buttons (Add to Cart and Favorite) on hover', () => {
    const container = screen.getByAltText('Test Product').closest('div');

    fireEvent.mouseOver(container!);

    expect(screen.getByRole('button', { name: /Add to Cart/i })).toBeVisible();
    expect(
      screen.getByRole('button', { name: /Favorite product/i }),
    ).toBeVisible();
  });

  it('handles Add to Cart button click', () => {
    const addToCartButton = screen.getByRole('button', {
      name: /Add to Cart/i,
    });
    fireEvent.click(addToCartButton);
    expect(mockOnAddToCart).toHaveBeenCalledTimes(0);
  });

  it('handles Favorite button click', () => {
    const favoriteButton = screen.getByRole('button', {
      name: /Favorite product/i,
    });

    fireEvent.click(favoriteButton);
    expect(mockOnFavorite).toHaveBeenCalledTimes(0);
  });

  it('renders without optional props', () => {
    renderResult.rerender(
      <ProductCard
        id="2"
        name="Minimal Product"
        images={['/minimal-image.jpg']}
        price="30"
        originalPrice={''}
      />,
    );

    expect(screen.getByText('Minimal Product')).toBeInTheDocument();
    expect(screen.queryByText('70')).not.toBeInTheDocument();
    expect(screen.queryByText('28% OFF')).not.toBeInTheDocument();
    expect(screen.getByText('Hot')).toBeInTheDocument();
  });
});
