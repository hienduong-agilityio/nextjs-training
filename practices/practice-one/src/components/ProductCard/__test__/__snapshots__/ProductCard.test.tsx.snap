// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col transition-shadow border-2 rounded-sm border-secondary-200 hover:shadow-lg"
  >
    <span
      class="absolute top-0 left-0 z-20 px-3 py-1 text-xs text-white uppercase rounded-sm bg-danger-200"
    >
      Hot
    </span>
    <div
      class="relative bg-secondary-200 group"
    >
      <img
        alt="Test Product"
        class="mix-blend-multiply object-contain h-[273px] w-[200px] sm:w-[300px] sm:h-[273px] mx-auto"
        data-nimg="1"
        decoding="async"
        height="0"
        loading="lazy"
        sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
        src="/_next/image?url=%2Ftest-image.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Ftest-image.jpg&w=256&q=75 256w, /_next/image?url=%2Ftest-image.jpg&w=384&q=75 384w, /_next/image?url=%2Ftest-image.jpg&w=640&q=75 640w, /_next/image?url=%2Ftest-image.jpg&w=750&q=75 750w, /_next/image?url=%2Ftest-image.jpg&w=828&q=75 828w, /_next/image?url=%2Ftest-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image.jpg&w=3840&q=75 3840w"
        style="color: transparent;"
        width="0"
      />
      <div
        class="absolute inset-0 z-10 flex items-center justify-center gap-4 mx-3 my-5 transition-opacity duration-300 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
      >
        <button
          aria-label="Favorite product"
          class="flex text-sm font-medium bg-indigo-600 border-2 rounded-full text-primary-100 border-primary-100"
          type="button"
        >
          <svg
            fill="none"
            height="35"
            viewBox="0 0 84 80"
            width="35"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(24.75, 24.75) scale(0.4375)"
            >
              <path
                clip-rule="evenodd"
                d="M54.285 9.72c2.857.003 5.652.816 8.05 2.341a14.652 14.652 0 0 1 5.467 6.258 14.344 14.344 0 0 1 1.136 8.171 14.466 14.466 0 0 1-3.56 7.469c-2.72 3.011-25.913 24.339-25.913 24.339S16.225 36.97 13.505 33.91a14.238 14.238 0 0 1-3.71-9.62 14.454 14.454 0 0 1 4.434-10.143 14.977 14.977 0 0 1 10.401-4.18c3.89 0 7.626 1.5 10.4 4.18a14.454 14.454 0 0 1 4.435 10.143c0-1.914.384-3.81 1.13-5.578a14.56 14.56 0 0 1 3.216-4.728 14.857 14.857 0 0 1 4.814-3.16 15.07 15.07 0 0 1 5.678-1.11l-.018.005ZM54.303 0a24.887 24.887 0 0 0-14.838 4.86A25.052 25.052 0 0 0 22.424.18C16.35.738 10.701 3.482 6.562 7.884 2.424 12.286.087 18.036.002 24.028-.082 30.02 2.09 35.83 6.103 40.344c2.778 3.13 21.072 19.954 26.622 25.037A9.988 9.988 0 0 0 39.481 68a9.987 9.987 0 0 0 6.755-2.62c5.52-5.077 23.738-21.858 26.551-24.983a24.106 24.106 0 0 0 5.932-12.45 23.902 23.902 0 0 0-1.897-13.62A24.418 24.418 0 0 0 67.704 3.9 25.05 25.05 0 0 0 54.285.004L54.303 0Z"
                fill="#33A0FF"
                fill-rule="evenodd"
              />
            </g>
          </svg>
        </button>
        <button
          aria-label="Add to Cart"
          class="text-sm font-medium bg-indigo-600 border-2 rounded-full text-primary-100 border-primary-100"
          type="button"
        >
          <svg
            fill="none"
            height="35"
            viewBox="0 0 90 85"
            width="35"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(29.6, 28.6) scale(1.5)"
            >
              <path
                clip-rule="evenodd"
                d="M15.22 12.705c.526 0 1.036-.18 1.442-.507.407-.326.684-.78.786-1.285L18.59 5.34a2.176 2.176 0 0 0-.023-.976 2.208 2.208 0 0 0-.449-.872 2.276 2.276 0 0 0-.787-.602 2.33 2.33 0 0 0-.975-.216H4.988V1.559c0-.295-.12-.578-.333-.786a1.156 1.156 0 0 0-.804-.328H1.582c-.297.009-.578.13-.785.338a1.101 1.101 0 0 0-.323.777c0 .29.116.568.323.776.207.208.488.33.785.338h1.142v11.145a2.323 2.323 0 0 0-1.271.37c-.377.245-.671.592-.845.999a2.18 2.18 0 0 0-.134 1.289c.086.433.303.83.622 1.144a2.3 2.3 0 0 0 1.169.612c.443.087.902.044 1.32-.125a2.272 2.272 0 0 0 1.026-.82c.251-.367.386-.799.386-1.24h7.953c.002.439.137.867.388 1.23.25.365.606.648 1.022.815.416.166.872.209 1.313.122.44-.086.844-.299 1.162-.61.317-.31.532-.706.62-1.136.086-.43.04-.877-.132-1.282a2.233 2.233 0 0 0-.837-.994 2.313 2.313 0 0 0-1.262-.374H4.996v-1.114H15.22Zm1.141-7.8-1.141 5.573H4.99V4.905H16.361Z"
                fill="#33A0FF"
                fill-rule="evenodd"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
    >
      <h3
        class="text-lg font-semibold text-center text-indigo"
      >
        Test Product
      </h3>
      <div
        class="flex gap-2 space-x-2"
      >
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.3181818181818182)"
          />
        </svg>
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.3181818181818182)"
          />
        </svg>
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.3181818181818182)"
          />
        </svg>
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.3181818181818182)"
          />
        </svg>
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#C1C8CE"
            transform="scale(0.3181818181818182)"
          />
        </svg>
      </div>
      <div
        class="flex items-center space-x-2"
      >
        <span
          class="text-lg font-medium text-primary-200"
        >
          $50
        </span>
        <span
          class="text-sm text-gray-400 line-through"
        >
          $70
        </span>
        <span
          class="text-sm font-semibold text-danger-50"
        >
          28% OFF
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
